src/components/Filters/index.tsx

```diff
@@ -81,291 +89,217 @@ const FilterWrapper = styled(Box)`
+		const item = currentEdit[index];
+		currentEdit.splice(index, 1, { ...item, value });
 		this.setState({
-			showModal: true,
-			edit: cloneDeep(rule),
+			edit: currentEdit,
 		});
 	}
 
-	removeRule(rule: FilterRule) {
-		if (rule.name === sieve.SIMPLE_SEARCH_NAME) {
-			this.setState({ searchString: '' });
-		}
-
-		const { rules } = this.props;
+	addFilter() {
+		const $id = this.state.editingFilter;
+		const { schema } = this.props;
+		const filter = SchemaSieve.createFilter(schema, this.state.edit);
+		const currentFilters = this.state.filters;
 
-		const updatedRules = rules.filter(r => r.id !== rule.id);
-		this.filterAndSetRules(updatedRules);
-	}
+		let filters: JSONSchema6[];
 
-	getDefaultEditData(data: Partial<FilterRule>, value: string) {
-		const update = cloneDeep(data);
-		const inputModels = sieve.makeFilterInputs(this.props.schema);
-		const model = inputModels[value];
-		update.type = model.type;
-		update.name = value;
-		update.operator = model.availableOperators[0].value;
-		update.label = model.label;
-		if (model.type === 'Date Time') {
-			update.value = moment().format('YYYY-MM-DDTHH:mm');
-		} else if (model.type === 'Date') {
-			update.value = moment().format('YYYY-MM-DD');
-		} else if (model.type === 'Time') {
-			update.value = moment().format('HH:mm');
-		} else if (model.type === 'Enum') {
-			update.value = first(this.props.schema[model.name].values);
+		if (!!$id) {
+			const matchIndex = findIndex(currentFilters, { $id });
+			currentFilters.splice(matchIndex, 1, filter);
+			filters = currentFilters.slice();
 		} else {
-			update.value = '';
+			filters = currentFilters.concat(filter);
 		}
 
-		return update;
+		console.log('FILTERS', filters);
+
+		this.setState(
+			{
+				filters,
+				edit: [this.getCleanEditModel()],
+				showModal: false,
+				editingFilter: null,
+			},
+			() =>
+				this.props.onFiltersUpdate &&
+				this.props.onFiltersUpdate(this.state.filters),
+		);
 	}
 
-	updateFilter(
-		filterBase: FilterModel,
-		attribute: keyof FilterModel,
-		value: any,
-	) {
-		let filter = { ...filterBase };
-
-		if (attribute === 'name' && filter.name !== value) {
-			filter = this.getDefaultEditData(filter, value) as FilterModel;
-		} else if (attribute === 'operator') {
-			filter.value = '';
-			filter[attribute] = value;
-		} else {
-			set(filter, attribute, value);
-		}
+	editFilter(filter: JSONSchema6) {
+		const { schema } = this.props;
 
-		return filter;
+		const signatures = SchemaSieve.decodeFilter(schema, filter);
+
+		this.setState({
+			edit: signatures as EditModel[],
+			editingFilter: filter.$id!,
+			showModal: true,
+		});
 	}
 
-	handleFilterFormChange(value: string, attribute: keyof FilterModel) {
-		const update = this.updateFilter(
-			this.state.edit as FilterModel,
-			attribute,
-			value,
+	removeFilter({ $id }: JSONSchema6) {
+		this.setState(
+			prevState => ({
+				filters: prevState.filters.filter(f => f.$id !== $id),
```

Comment:
How about [`_.reject`](http://devdocs.io/lodash~4/index#reject)?
(filtering the filters!!!)

Reply:
I always forget about reject :) it's so neat